funsor/minipyro.py

```diff
@@ -240,35 +241,35 @@ def sample(name, fn, obs=None):
 
 # param is an effectful version of PARAM_STORE.setdefault
 # When any effect handlers are active, it constructs an initial message and calls apply_stack.
-def param(name, init_value=None, event_dim=None):
+def param(name, init_value=None, constraint=None, event_dim=None):
     cond_indep_stack = {}
     output = None
     if init_value is not None:
         if event_dim is None:
             event_dim = init_value.dim()
         output = funsor.reals(*init_value.shape[init_value.dim() - event_dim:])
+    if constraint is None:
+        constraint = torch.distributions.constraints.real
 
-    def fn(init_value):
-        if name in PARAM_STORE:
+    def fn(init_value, constraint):
```

Comment:
The following logic was adapted from [ParamStoreDict.get_param()](https://github.com/pyro-ppl/pyro/blob/dbe0415/pyro/params/param_store.py#L186).